Automattic/woocommerce-payments
# Fix - UPE Appearance on WC Blocks

Fixes #4145 

#### Changes proposed in this Pull Request

Classic checkout automatically generates styles for the UPE payment form directly form the checkout page. However when using Blocks checkout this doesn't work because the appearance value was passed to the `PaymentElement` and appearance data was not generated since the selectors used on Blocks checkout are different.

This PR refactors the existing functions to generate UPE styles to allow style generation for Blocks checkout and adds the styles to the [Elements Appearance API](https://stripe.com/docs/stripe-js/appearance-api) instead of `PaymentElement`.

#### Testing instructions

* Enable UPE Payment methods & WC Blocks on the test site.
* On `develop` add a product to cart and go to classic checkout. Notice that the UPE Payment form styles match the input styles on the checkout page.
* Go to Blocks checkout and notice that the UPE payment form uses default styles and doesn't match the input styles on the page
* Checkout the branch `fix/upe-blocks-appearance` and rebuilt assets
* Go to classic checkout and confirm that the UPE payment form styles matches input elements on the page
* Go to Blocks checkout and confirm that the UPE payment form styles matches input elements on the page.

**Notes**
* The label transitions on Blocks checkout input fields are not passed to the payment form
* There is a workaround in place to rewrite `text-indent` used by WC Blocks to `padding`. WC Blocks introduced using `text-indent` on this [PR](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5947) and stripe doesn't support this property.

**Further Optimizations**

The styles for payment form on Blocks checkout like styles for invalid labels & validation messages can be optimized further on a different PR.

-------------------


**Post merge**
